src/eviction/eviction.c

```diff
@@ -87,20 +91,35 @@ static inline uint32_t ocf_evict_do(ocf_cache_t cache,
 			/* It seams that no more partition for eviction */
 			break;
 		}
-		if (evicted >= evict_cline_no) {
-			/* Evicted requested number of cache line, stop */
-			goto out;
-		}
 
-		to_evict = ocf_evict_calculate(cache, part, evict_cline_no,
-				true);
+		/* in first loop iteration only consider overflown partitions */
+		overflow_size = ocf_part_overflow_size(cache, part);
+		if (overflown && overflow_size == 0)
+			continue;
+
+		to_evict = ocf_evict_calculate(cache, part,
+				evict_cline_no - evicted, true);
 		if (to_evict == 0) {
 			/* No cache lines to evict for this partition */
 			continue;
 		}
 
+		if (overflown)
+			to_evict = OCF_MIN(to_evict, overflow_size);
+
 		evicted += ocf_eviction_need_space(cache, io_queue,
 				part, to_evict);
+
+		if (evicted >= evict_cline_no) {
+			/* Evicted requested number of cache line, stop */
+			goto out;
+		}
+
+	}
+
+	if (overflown) {
+		overflown = false;
+		goto repeat;
```

Comment:
Maybe.. let's change it up to make a proper loop? I understand having `goto`'s as error handling, but having it instead of loops is too much for me

Reply:
done